### PR TITLE
[RDY] vim-patch:7.4.699

### PR DESCRIPTION
**Pull request by grtlr**
```
Problem:    E315 when trying to delete a fold. (Yutao Yuan)
Solution:   Make sure the fold doesn't go beyond the last buffer line.
            (Christian Brabandt)
```

https://code.google.com/p/vim/source/detail?r=v7-4-699

Original patch:

``` diff
  diff --git a/src/nvim/fold.c b/src/nvim/fold.c
--- a/src/nvim/fold.c
+++ b/src/nvim/fold.c
@@ -234,6 +234,8 @@ hasFoldingWin(win, lnum, firstp, lastp, 
        return FALSE;
     }

+    if (last > win->w_buffer->b_ml.ml_line_count)
+       last = win->w_buffer->b_ml.ml_line_count;
     if (lastp != NULL)
        *lastp = last;
     if (firstp != NULL)
diff --git a/src/nvim/version.c b/src/nvim/version.c
--- a/src/nvim/version.c
+++ b/src/nvim/version.c
@@ -742,6 +742,8 @@ static char *(features[]) =
 static int included_patches[] =
 {   /* Add new patch number below this line */
 /**/
+    699,
+/**/
     698,
 /**/
     697,
```

### CODE REVIEW

File: src/nvim/fold.c [hunk 1]
```diff
@@ -154,7 +154,7 @@ int hasFolding(linenr_T lnum, linenr_T *firstp, linenr_T *lastp)
 }
 
 /* hasFoldingWin() {{{2 */
-int 
+int
```
Comment (glts) [inline src/nvim/fold.c]: I wouldn't include this unrelated change.

File: src/nvim/fold.c [hunk 2]
```diff
@@ -241,6 +241,8 @@ hasFoldingWin (
     return FALSE;
   }
 
+  if (last > win->w_buffer->b_ml.ml_line_count)
```
Comment (glts) [inline src/nvim/fold.c]: Always use curly brackets.

**Comment by glts** [comment]
Other than that, looks good to me. :+1:

File: src/nvim/fold.c (hunk 2 above)
Comment (grtlr) [inline src/nvim/fold.c]: Should I also change to curly brackets in the 2 if statements below?

Comment (glts) [inline src/nvim/fold.c]: Personally I wouldn't, but I guess it's a matter of opinion. I think the rule of thumb is to keep the commits focused so that the history is as clean as possible and `git blame`/`:Gblame` is as useful as possible.

File: src/nvim/fold.c (hunk 1 above)
Comment (Pyrohh) [inline src/nvim/fold.c]: :+1:

**Comment by grtlr** [comment]
Added curly brackets. Thanks for the feedback!

**Comment by Pyrohh** [comment]
I'm probably doing something wrong, but I can't to reproduce this. Have either of you managed to do so?

**Comment by glts** [comment]
@Pyrohh Yes, first message in https://groups.google.com/d/msg/vim_dev/Kt_PW54CMP0/8u_WwYErJ2cJ has repro steps.

**Comment by Pyrohh** [comment]
Merged, thanks for the link @glts. I was mistakenly looking at Vim's issue tracker and not vim-dev.

**Comment by Pyrohh** [comment]
Just as a note, the Travis build was passing before @grtlr rebased (just cosmetic changes IIRC). Still, I had forgotten to wait for Travis, which I'll make sure to do so in the future.
